app/routes.test.js

```diff
@@ -95,266 +29,50 @@ describe('GET /healthcheck', () => {
-        expect(res.text.includes(`<h2 data-test-id="section-title">${mockSectionErrorContext.title}</h2>`)).toEqual(true);
+        expect(res.text.includes('<h1>Preview Page</h1>')).toEqual(true);
         expect(res.text.includes('data-test-id="error-page-title"')).toEqual(false);
-        sectionControllers.getSectionPageErrorContext.mockReset();
-      });
-  });
-
-  it('should return error page if there is an api error', () => {
-    sectionControllers.postSection = jest.fn()
-      .mockImplementation(() => Promise.reject(new Error('er')));
-    const app = new App().createApp();
-    app.use('/', routes);
-    return request(app)
-      .post('/solution/1/dashboard/a-dashboard/section/a-section')
-      .send({})
-      .expect(200)
-      .then((res) => {
-        expect(res.text.includes(`<h2 data-test-id="section-title">${mockSectionContext.title}</h2>`)).toEqual(false);
-        expect(res.text.includes('data-test-id="error-page-title"')).toEqual(true);
       });
   });
 });
 
-describe('GET /solution/:solutionId/preview', () => {
-  afterEach(() => {
-    previewControllers.getPreviewPageContext.mockReset();
-  });
-
-  it('should return the correct status and text if there is no error', () => {
-    previewControllers.getPreviewPageContext = jest.fn()
-      .mockImplementation(() => Promise.resolve(mockPreviewContext));
+describe('Error handler', () => {
+  it('should return error page if there is an error from /preview', () => {
+    previewControllers.getPreviewPageContext = jest.fn().mockImplementation(() => Promise.reject());
     const app = new App().createApp();
     app.use('/', routes);
     return request(app)
       .get('/solution/1/preview')
       .expect(200)
       .then((res) => {
-        expect(res.text.includes('<h1>Preview Page</h1>')).toEqual(true);
-        expect(res.text.includes('data-test-id="error-page-title"')).toEqual(false);
+        expect(res.text.includes('<h1>Preview Page</h1>')).toEqual(false);
+        expect(res.text.includes('data-test-id="error-page-title"')).toEqual(true);
       });
   });
 
-  it('should return error page if there is an error', () => {
-    previewControllers.getPreviewPageContext = jest.fn().mockImplementation(() => Promise.reject());
+  it('should return error page if there is an error from the supplier route', () => {
+    dashboardControllers.getMarketingPageDashboardContext = jest.fn()
+      .mockImplementation(() => Promise.reject());
     const app = new App().createApp();
     app.use('/', routes);
     return request(app)
-      .get('/solution/1')
+      .get('/supplier/solution/1')
       .expect(200)
       .then((res) => {
-        expect(res.text.includes('<h1>Preview Page</h1>')).toEqual(false);
+        expect(res.text.includes('data-test-id="dashboard"')).toEqual(false);
         expect(res.text.includes('data-test-id="error-page-title"')).toEqual(true);
       });
   });
```

Comment:
do you think it is worth writing tests to check the supplier router is being called?

Reply:
👍 will add a test